tsl/src/continuous_aggs/create.c

```diff
@@ -1147,11 +1148,13 @@ get_bucket_width(CAggTimebucketInfo bucket_info)
 				bucket_info.interval->day = bucket_info.interval->month * DAYS_PER_MONTH;
 				bucket_info.interval->month = 0;
 			}
+
 			Datum epoch = DirectFunctionCall2(interval_part,
 											  PointerGetDatum(cstring_to_text("epoch")),
 											  IntervalPGetDatum(bucket_info.interval));
+
 			/* Cast float8 to int8. */
-			width = DatumGetInt64(DirectFunctionCall1(dtoi8, epoch));
+			width = DatumGetFloat8(epoch);
```

Comment:
Why are we not converting to milliseconds and using integer comparisons?
floating point comparisons are not reliable, subject to rounding error etc.

Reply:
thank you @gayyappan for the review! Converting to millis/micros is what I'd initially attempted, but I don't have a strong opinion...

Reply:
@gayyappan are u talking about using `ts_interval_value_to_internal` function??

Reply:
I haven't looked into how to implement this. My observation is that using floats for comparisons will have rounding errors. It is safer to convert everything to int64 and compare values as integers.  Yes, this function looks like a good candidate to convert the interval as an integer value.

Reply:
You are correct @gayyappan ... it can be even simpler @konskov:
```diff
diff --git a/tsl/src/continuous_aggs/create.c b/tsl/src/continuous_aggs/create.c
index ada8166d1..8645630b8 100644
--- a/tsl/src/continuous_aggs/create.c
+++ b/tsl/src/continuous_aggs/create.c
@@ -1151,11 +1151,9 @@ get_bucket_width(CAggTimebucketInfo bucket_info)
                bucket_info.interval->day = bucket_info.interval->month * DAYS_PER_MONTH;
                bucket_info.interval->month = 0;
            }
-           Datum epoch = DirectFunctionCall2(interval_part,
-                                             PointerGetDatum(cstring_to_text("epoch")),
-                                             IntervalPGetDatum(bucket_info.interval));
-           /* Cast float8 to int8. */
-           width = DatumGetInt64(DirectFunctionCall1(dtoi8, epoch));
+
+           /* Convert Interval to int64 */
+           width = ts_interval_value_to_internal(IntervalPGetDatum(bucket_info.interval), INTERVALOID);
            break;
        }
        default:
```